lib/cast/int32.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const assert = require('assert');
+const BSON = require('bson');
+
+/**
+ * Given a value, cast it to a Int32, or throw an `Error` if the value
+ * cannot be casted. `null` and `undefined` are considered valid.
+ *
+ * @param {Any} value
+ * @return {Number}
+ * @throws {Error} if `value` does not represent an integer, or is outside the bounds of an 32-bit integer.
+ * @api private
+ */
+
+module.exports = function castInt32(val) {
+  if (val == null) {
+    return val;
+  }
+  if (val === '') {
+    return null;
+  }
+
+  const coercedVal = val instanceof BSON.Long ? val.toNumber() : Number(val);
```

Comment:
https://github.com/mongodb-js/mongoose/pull/1#discussion_r1847157308

Shouldn't be using instanceof here per convo ^

Reply:
My bad! Fixed now!

Reply:
Oh I see! I'll change it again.